src/components/team/billing-section.tsx

```diff
@@ -72,14 +75,31 @@ const BillingSection = ({
   const activeSubscription =
     get(subscriptionData, 'subscription.status') === 'active' &&
     !get(subscriptionData, 'subscription.canceled_at')
-  const nextBillDate: string | undefined = activeSubscription
-    ? format(
-        new Date(
-          get(subscriptionData, 'upcomingInvoice.next_payment_attempt') * 1000,
-        ),
-        'yyyy/MM/dd',
-      )
-    : undefined
+
+  // Figure out what to display for the Next Billing Date.
+  //
+  // If it is too early in the billing period or if the subscription was just
+  // created, there may not be an upcoming invoice generated yet. In that case,
+  // we see the subscription is active and display 'Pending' for now.
+  const nextPaymentAttempt = get(
+    subscriptionData,
+    'upcomingInvoice.next_payment_attempt',
+  )
+  let nextBillDateDisplay: string
+  if (!activeSubscription) {
+    nextBillDateDisplay = 'Canceled'
+  } else if (activeSubscription && isEmpty(subscriptionData.upcomingInvoice)) {
+    nextBillDateDisplay = 'Pending'
+  } else if (activeSubscription && nextPaymentAttempt) {
+    nextBillDateDisplay = format(
+      new Date(
+        get(subscriptionData, 'upcomingInvoice.next_payment_attempt') * 1000,
+      ),
+      'yyyy/MM/dd',
+    )
+  } else {
+    nextBillDateDisplay = '-'
+  }
```

Comment:
I'm not super excited about adding this big if/else statement in, but 1) we don't always have the upcoming invoice and 2) we want to always display _something_ instead of erroring out on invalid dates or a missed edge case.